### PR TITLE
[WIP] spawning and player list

### PR DESCRIPTION
**Pull request by ghost**
still crashes iOS clients and doesn't spawn the player

### CODE REVIEW

**Comment by ghost** [comment]
currently, tab works perfectly, players are removed and added. fixed removing players from the main player array. an example player is spawned but other players still don't show up. further investigation is required

File: lib/plugins/login.js [hunk 1]
```diff
@@ -10,15 +10,14 @@ const World = require('../world');
...
-  // send a message to all players that the server has been shut down
-  process.on('SIGINT', function() {
-    serv._writeAll('disconnect', {
-      message: 'Server closed'
-    });
-    setTimeout(function(){
-      process.exit(0);
-    }, 1000);
-  });
+  // process.on('SIGINT', function() {
```
Comment (rom1504) [inline lib/plugins/login.js:13]: why did you remove this ?

File: lib/plugins/spawn.js [hunk 2]
```diff
@@ -1,28 +1,78 @@
 'use strict';
 var Vec3 = require('vec3');
+var uuid = require('node-uuid');
```
Comment (rom1504) [inline lib/plugins/spawn.js:3]: we don't need this

File: lib/plugins/spawn.js [hunk 3]
```diff
@@ -101,8 +158,91 @@ module.exports.player = function(player, serv){
         time: 0,
         started: 1
       });
+
+      player.spawned = true;
     });
 
+    setTimeout(function() {
```
Comment (rom1504) [inline lib/plugins/spawn.js:165]: why is the setTimeout needed oO ?

File: lib/plugins/spawn.js [hunk 4]
```diff
@@ -101,8 +158,91 @@ module.exports.player = function(player, serv){
...
+      serv._writeAll('player_list', {
+        type: 0,
+        entries: serv.list
+      });
+      serv.players.forEach(function(target) {
+
```
Comment (rom1504) [inline lib/plugins/spawn.js:171]: much duplication here

**Comment by rom1504** [comment]
I made a few comments, have a look ;)

Overall, nice !
